File: vm/context.go
package vm

import "context"

type Context interface {
	// Context returns the go context of the function instance
	Context() context.Context

	// Close calls the go context cancel method.
	Close() error

	// Project returns the Taubyte project id
	Project() string

	// Application returns the application, if none returns an empty string
	Application() string

	// Resource returns the id of the resource being used.
	Resource() string

	// Branch returns the branch name used by this resource execution pipeline.
	Branch() string

	// Commit returns the commit id used by this resource execution pipeline.
	Commit() string
}

File: rust/vm/context.rs
use std::sync::Arc;
use async_trait::async_trait;

/// A trait representing the context interface in Rust
#[async_trait]
pub trait Context: Send + Sync {
    /// Returns the Rust async runtime context (`tokio::context` equivalent).
    fn context(&self) -> Arc<tokio::sync::RwLock<tokio::task::JoinHandle<()>>>;

    /// Returns the project ID.
    fn project(&self) -> String;

    /// Returns the application name, or an empty string if none exists.
    fn application(&self) -> String;

    /// Returns the resource ID being used.
    fn resource() -> ID(Clone);

    /// Implement  branch function aspects parsing`]!("{name}=Options...")
    }

    
    
    

